internal/executor/docker.go

```diff
@@ -264,25 +264,33 @@ func (e *docker) Wait() error {
 
 	containerID := e.getContainerID()
 
-	waitDone, errs := e.client.ContainerWait(e.ctx, containerID, container.WaitConditionNotRunning)
+	waitCtx, waitCancel := context.WithCancel(context.Background())
```

Comment:
I changed this to use just `e.ctx`.

I originally wanted to use a context separate from  `e.ctx` because if the process was killed by context timeout, it was hitting a case where the context would be done, the container would exit, and the select would randomly pick between the `e.ctx.Done()` case and the `waitDone` case. `ContainerWait` would return early anyways once `e.ctx` was cancelled since the container was killed. This extra context shouldn't be necessary anymore because I handled the `e.ctx` cancellation specially in both cases.